### PR TITLE
Improves imagenet examples file structure

### PR DESCRIPTION
**Pull request by andsteing**
Adheres to new filestructure from #231 (#480).

### CODE REVIEW

File: linen_examples/imagenet/main.py [hunk 1]
```diff
@@ -37,6 +38,10 @@
     'model_dir', default=None,
     help=('Directory to store model data.'))
 
+flags.DEFINE_string(
```
Comment (Marvin182) [inline linen_examples/imagenet/main.py:41]: Can we use the TFDS_DATA_DIR environment variable instead?

File: linen_examples/imagenet/main.py [hunk 2]
```diff
@@ -37,6 +38,10 @@
     'model_dir', default=None,
```
Comment (Marvin182) [inline linen_examples/imagenet/main.py:38]: Other examples use --workdir instead of --model_dir. Can we converge to --workdir everywhere?

**Review by Marvin182 — CHANGES_REQUESTED** [review]
LGTM apart from the flags.

File: linen_examples/imagenet/main.py (hunk 2 above)
Comment (andsteing) [reply linen_examples/imagenet/main.py]: Done. Also updated https://github.com/google/flax/issues/231#issuecomment-716568710

**Pull request merged**